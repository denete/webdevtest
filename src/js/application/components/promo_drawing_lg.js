import React from "react";
import { Link } from "react-router";
import dateFormat from "dateformat"

// large format for drawing schedule table
export default class PromoDrawingLg extends React.Component {
  render(){
    const drawings = this.props.drawings;

    const rows = drawings.map((o,i)=>
      <tr key={i}>
        <td>{o.prize}</td>
        <td>{dateFormat(o.entry_deadline, "fullDate")}</td>
        <td>{dateFormat(o.drawing_date, "fullDate")}</td>
      </tr>
    )

    return (
        <div class="drawing_sched_lg">
          <h2 class="drawing_schedule_title">Drawing Schedule</h2>
          <table>
            <tbody>
              <tr>
                <th>Prize</th>
                <th>Entry Deadline</th>
                <th>Drawing Date</th>
              </tr>
              {rows}
            </tbody>
          </table>
        </div>
    );
  }
}
