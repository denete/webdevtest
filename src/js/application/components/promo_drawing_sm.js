import React from "react";
import { Link } from "react-router";
import dateFormat from "dateformat";

// small format for drawing schedule table
export default class PromoDrawingSm extends React.Component {
  render(){
    const drawings = this.props.drawings;

    const rows = drawings.map((o,i)=>
      <table key={i}>
        <tbody>
          <tr>
            <th>Prize</th>
            <td>{o.prize}</td>
          </tr>
          <tr>
            <th>Entry Deadline</th>
            <td>{dateFormat(o.entry_deadline, "fullDate")}</td>
          </tr>
          <tr>
            <th>Drawing Date</th>
            <td>{dateFormat(o.drawing_date, "fullDate")}</td>
          </tr>
        </tbody>
      </table>
    )

    return (
        <div class="drawing_sched_sm">
          <h2 class="drawing_schedule_title">Drawing Schedule</h2>
          {rows}
        </div>
    );
  }
}
