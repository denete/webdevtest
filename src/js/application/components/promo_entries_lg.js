import React from "react";
import { Link } from "react-router";
import dateFormat from "dateformat";

// table for large format Ticket entries
export default class PromoEntriesLg extends React.Component {
  render(){
    const entries = this.props.entries;

    const rows = entries.map((o,i)=>
      <tr key={i}>
        <td>{o.entry_number}</td>
        <td>{dateFormat(o.date, "fullDate")}</td>
      </tr>
    )

    return (
      <div class="ticket_entries_lg">
        <h2 class="total_entered">Your total Tickets Entered: {entries.length}</h2>
        <p>All Entries are locked in at the time they are submitted and cannotbe deleted.</p>
        <table>
          <thead>
            <tr>
              <th>Entry Number</th>
              <th>Date</th>
            </tr>
          </thead>
          <tbody>
            {rows}
          </tbody>
        </table>
      </div>
    );
  }
}
