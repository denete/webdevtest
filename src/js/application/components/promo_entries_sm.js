import React from "react";
import { Link } from "react-router";
import dateFormat from "dateformat";

// table for small format Ticket entries
export default class PromoEntriesSm extends React.Component {
  render(){
    const entries = this.props.entries;

    const rows = entries.map((o,i)=>
      <table key={i}>
        <tbody>
          <tr>
            <th>Entry Number</th>
            <td>{o.entry_number}</td>
          </tr>
          <tr>
            <th>Date</th>
            <td>{dateFormat(o.date, "fullDate")}</td>
          </tr>
        </tbody>
      </table>
    )

    return (
      <div class="ticket_entries_sm">
        <h2 class="total_entered">Your total Tickets Entered: {entries.length}</h2>
        <p>All Entries are locked in at the time they are submitted and cannotbe deleted.</p>
        {rows}
      </div>
    );
  }
}
