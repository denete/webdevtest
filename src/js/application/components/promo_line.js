import React from "react";
import { Link } from "react-router";
import dateFormat from "dateformat";

// Each promotion line
export default class PromoLine extends React.Component {

  render(){
    const promo_item = this.props.promo_item;
    const promo_index = this.props.promo_index;

    return (
      <div class="promo_line">
        <div class="">
          <img src={"./" + promo_item.promo_image_url} alt=""/>
          <Link to={"/promotion/" + promo_index}><h2>{promo_item.promotion_name}</h2></Link>
          <p>{promo_item.summary}</p>
          <p>Next Drawing Date: {dateFormat(promo_item.drawings[0].drawing_date, "fullDate")} </p>
        </div>
      </div>
    );
  }
}
