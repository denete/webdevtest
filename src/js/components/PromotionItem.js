import React from "react";

import moment from "moment";

import { injectIntl, defineMessages } from "react-intl";

import { Table, Thead, Tbody, Tr, Th, Td } from "react-super-responsive-table";

import "react-super-responsive-table/dist/SuperResponsiveTableStyle.css";

const messages = defineMessages({
	drawing_schedule: {
		id: "DRAWING_SCHEDULE",
		defaultMessage: "Drawings schedule"
	},
	total_tickets_entered: {
		id: "TOTAL_TICKETS_ENTERED",
		defaultMessage: "Your total tickets entered"
	},
	entry_description: {
		id: "ENTRY_DESCRIPTION",
		defaultMessage: "All entries are locked in at the time they are submitted and cannot be deleted."
	}
});

const drawingHeaders = ["Prize", "Entry deadline", "drawing date"];

const entryHeaders = ["Entry number", "date"];

class PromotionItem extends React.Component {
    renderDrawingSchedule (promotionData) {
        const { intl } = this.props;

        const drawingsTHeadContent = drawingHeaders.map((drawingHeader, index) => {
            const headerKey = `header${index}`;
            return <Th key={ headerKey }>{ drawingHeader }</Th>;
        });

        const drawingsTBodyContent = promotionData.drawings.map((drawingData, index) => {
            const key = `drawings${index}`;
            const deadlineDate = moment(drawingData.entry_deadline).format("dddd, MMMM DD, YYYY");
            const drawingDate = moment(drawingData.drawing_date).format("dddd, MMMM DD, YYYY");
            return (
                <Tr key={key}>
                    <Td>{drawingData.prize}</Td>
                    <Td>{deadlineDate}</Td>
                    <Td>{drawingDate}</Td>
                </Tr>
            );
        });

        const drawingScheduleText = intl.formatMessage(messages.drawing_schedule);
        return (
            <div>
                <p>{ drawingScheduleText }</p>
                <Table className="tableResults">
                    <Thead>
                        <Tr>{ drawingsTHeadContent }</Tr>
                    </Thead>

                    <Tbody>
                        { drawingsTBodyContent }
                    </Tbody>
                </Table>
            </div>
        );
    }

    renderEntries (promotionData) {
        const { intl } = this.props;

        const entriesTHeadContent = entryHeaders.map((entryHeader, index) => {
            const headerKey = `header${index}`;
            return <Th key={ headerKey }>{ entryHeader }</Th>;
        });

        const entriesTBodyContent = promotionData.entries.map((entryData, index) => {
            const key = `entries${index}`;
            const entryDate = moment(entryData.date).format("dddd, MMMM DD, YYYY");
            return (
                <Tr key={key}>
                    <Td>{entryData.entry_number}</Td>
                    <Td>{entryDate}</Td>
                </Tr>
            );
        });

        const totalTicketsEnteredText = intl.formatMessage(messages.total_tickets_entered);
        const entryDescriptionText = intl.formatMessage(messages.entry_description);
        return (
            <div>
                <p>{ totalTicketsEnteredText } { promotionData.entries.length }</p>
                <p>{ entryDescriptionText }</p>
                <Table className="tableResults">
                    <Thead>
                        <Tr>
                            { entriesTHeadContent }
                        </Tr>
                    </Thead>

                    <Tbody>
                        { entriesTBodyContent }
                    </Tbody>
                </Table>
            </div>
        );
    }

    render () {
        const { promotionData } = this.props;

        return (
            <div>
                <img src={promotionData.promo_image_url}></img>
                <div>{promotionData.promotion_name}</div>
                { this.renderDrawingSchedule(promotionData) }
                <p>{promotionData.entry_info}</p>
                { this.renderEntries(promotionData) }
            </div>
        );
    }
}

export default injectIntl(PromotionItem);