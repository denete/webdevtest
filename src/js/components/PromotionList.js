import React from "react";
import numeral from "numeral";
import PromotionListItem from "./PromotionListItem";

class PromotionList extends React.Component {
    render () {
        const { promotion_objects: promotionsData, server_time: serverTimeData } = this.props.data;
        const promotionListItems = promotionsData.map((promotionData, index) => {
            const key = `promotionListItem${index}`;
            const url = `?promo=promo${numeral(index + 1).format("00")}`;
            return (
                <PromotionListItem
                    key={key}
                    url={url}
                    serverTimeData={serverTimeData}
                    promotionData={promotionData}
                />
            );
        });
        return (
            <div className="promotionListView">
                <div className="promotionList">
                    { promotionListItems }
                </div>
            </div>
        )
    }
}

export default PromotionList;