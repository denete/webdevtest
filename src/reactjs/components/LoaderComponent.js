import React from "react";

import { injectIntl, defineMessages } from "react-intl";

const messages = defineMessages({
	loading: {
		id: "LOADING",
		defaultMessage: "Loading..."
	}
});

export class LoaderComponent extends React.Component {
    render () {
        const loadingText = this.props.intl.formatMessage(messages.loading);
        return (
            <div className="modal">
              <div className="modalInner">
                { loadingText }
              </div>
            </div>
        );
    }
}

export default injectIntl(LoaderComponent);