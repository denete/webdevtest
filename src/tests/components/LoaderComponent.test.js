import React from 'react';
import { shallow } from 'enzyme';
import { nodeWithIntlProp } from "./../IntlTestHelper";
import { LoaderComponent } from "./../../reactjs/components/LoaderComponent";

test('should render LoaderComponent correctly', () => {
    const wrapper = shallow(nodeWithIntlProp(<LoaderComponent />));
    expect(wrapper).toMatchSnapshot();
});