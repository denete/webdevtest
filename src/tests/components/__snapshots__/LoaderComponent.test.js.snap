// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoaderComponent correctly 1`] = `
<div
  className="modal"
>
  <div
    className="modalInner"
  >
    Loading...
  </div>
</div>
`;
